### DISCUSSION TITLE
[Chapter 07] 스프링 부트 자동 설정의 선별 과정?

### DISCUSSION THREAD

**Discussion by Irisation23** [cowaine/spring-boot-for-msa#66] [Chapter 07. Spring AOP, Testing, and Auto Confiugration] [answered]
## 질문 개요

7.3 에 작성된 내용 중
```
스프링 부트는 자동 설정 기능이 있어 프레임워크에서 사용하는 기능들을 가장 보편적인 방법으로 미리 설정해서
사용자에게 제공한다.
```

와 같은 내용이 있다.

여기서 위의 글을 읽어보면 `가장 보편적인 방법` 이라는 키워드가 등장한다.

---

### Q. 스프링 부트 자동 설정에서 default 설정이 되어 있는 여러 설정의 가장 보편적인 방법의 선정 기준은 무엇일까?

- 가장 보편적인 방법의 예로 들면은 maven repository에서 이용하고 있는 dependency의 다운로드 횟수를 참조 하는 것일까?? 등등

**Comment by BvrPark** [comment, marked as answer]
스프링 부트는 주로 spring-boot-stater 패키지를 통해 기본 설정을 한다고 합니다.
그래서 starter 패키지가 여러가지 다른 라이브러리들을 모아놓은 것인데 선정 기준은 제 생각엔 
1. 인기도 
2. 사후 지원 
3. 기능성 

이렇게 중점으로 두고 default 설정을 하지 않을까 생각이 듭니다.

**Comment by corock** [comment]
제 생각으로는 "보편적인" 이라는 워딩을 사용한 것부터 저자도 명확하게 답변내릴 수 없는 내용이지 않나 조심스럽게 생각합니다.
결국에는 구현체를 만드는 개발자가 생각하는 보편적인 방법이 곧 보편적인 방법이 되지 않을까요? ㄷㄷ

**Comment by JoisFe** [comment]
https://docs.spring.io/spring-boot/docs/current/reference/html/using.html#using.auto-configuration

- 선정기준은 나와있지 않은 것 같네요..
